client/src/App.tsx

```diff
@@ -1,90 +1,81 @@
+        } else {
+          toast.error(`Deployed ${imageName} but no route was created.`);
+        }
+      }
+    });
     try {
-      output = await deployImage(imageName);
-      setDeployResponse(output);
-    } catch (err) {
-      toast.error(`Failed to deploy ${imageName}`);
-      const e = err as any;
-      output = output + '\n' + (e.stderr ? e.stderr : e);
-      setDeployResponse(output);
-      return;
+      deployer.deploy(imageName);
+    } catch (error) {
+      toast.error(getMessage(error));
+      setDeployResponse(output += `${getMessage(error)}\n`);
     }
-    const inspectionData = await getLocalImageInspectionJson(selectedImage.name);
-    let exposedPorts = inspectionData?.[0].Config.ExposedPorts;
+  }
 
-    const hasExposedPorts = exposedPorts && Object.keys(exposedPorts).length > 0;
-    if (hasExposedPorts) {
-      //Nothing to expose
-      const appName = getAppName(imageName);
-      try {
-        output = output + '\n' + await exposeService(appName);
-        setDeployResponse(output);
-      } catch (e) {
-        toast.error(`Failed to expose '${appName}' for  ${imageName}`);
-        output = output + '\n' + (e as any).stderr;
-        setDeployResponse(output);
-      }
-      const route = await getProjectRoute(appName);
-      if (route) {
-        toast.success(`Deployed ${imageName} to ${route}.`);
-        output += '\nApplication is exposed as: ' + route;
-        output += '\nWaiting for ' + route + ' to be ready.\n'
-        setDeployResponse(output);
-        let waitRoute = route;
-        if (process.env.NODE_ENV === 'development') {
-          waitRoute = `${process.env.REACT_APP_CORS_PROXY_URL}/${route}`;
-        }
-        await waitOnUrl(`${waitRoute}`, WAITING_ON_URL_TIMEOUT, 1000, (out) => {
-          output += out;
-          setDeployResponse(output);
-        }).then((response) => {
-          openInBrowser(route);
-          output += `\nApplication URL ${route} opened in browser`;
-          setDeployResponse(output);
-        }).catch((err) => {
-          if (err) {
-            console.log(err);
-            const isNotAccessibleMessage = `Application URL ${route} is not accessible.`;
-            output += `\n${isNotAccessibleMessage}`;
-            toast.warning(isNotAccessibleMessage);
-          } else {
-            const isStillNotAccessibleMessage = `Application URL ${route} is still not accessible after ${WAITING_ON_URL_TIMEOUT / 1000} seconds.`;
-            output += `\n${isStillNotAccessibleMessage}`;
-            toast.warning(isStillNotAccessibleMessage);
-          }
-          setDeployResponse(output);
-        });
+  async function waitAndOpen(route: string, currentOutput: string) {
+    let waitRoute = route;
+    if (process.env.NODE_ENV === 'development') {
+      waitRoute = `${process.env.REACT_APP_CORS_PROXY_URL}/${route}`;
+    }
+    let output = currentOutput;
+    await waitOnUrl(`${waitRoute}`, WAITING_ON_URL_TIMEOUT, 1000, (out) => {
```

Comment:
`${waitRoute}` should be just waitRoute.

Reply:
fixed